metrics-graphite/src/main/java/io/dropwizard/metrics/graphite/GraphiteReporter.java

```diff
@@ -223,63 +225,116 @@ public void stop() {
-        graphite.send(prefix(name, "count"), format(meter.getCount()), timestamp);
-        graphite.send(prefix(name, "m1_rate"),
-                      format(convertRate(meter.getOneMinuteRate())),
-                      timestamp);
-        graphite.send(prefix(name, "m5_rate"),
-                      format(convertRate(meter.getFiveMinuteRate())),
-                      timestamp);
-        graphite.send(prefix(name, "m15_rate"),
-                      format(convertRate(meter.getFifteenMinuteRate())),
-                      timestamp);
-        graphite.send(prefix(name, "mean_rate"),
-                      format(convertRate(meter.getMeanRate())),
-                      timestamp);
+        if (filter.matches(name, meter, "count")) {
+            graphite.send(prefix(name, "count"), format(meter.getCount()), timestamp);
+        }
+        if (filter.matches(name, meter, "m1_rate")) {
+            graphite.send(prefix(name, "m1_rate"),
+                    format(convertRate(meter.getOneMinuteRate())),
+                    timestamp);
+        }
+        if (filter.matches(name, meter, "m5_rate")) {
+            graphite.send(prefix(name, "m5_rate"),
+                    format(convertRate(meter.getFiveMinuteRate())),
+                    timestamp);
+        }
+        if (filter.matches(name, meter, "m15_rate")) {
+            graphite.send(prefix(name, "m15_rate"),
+                    format(convertRate(meter.getFifteenMinuteRate())),
+                    timestamp);
+        }
+        if (filter.matches(name, meter, "mean_rate")) {
+            graphite.send(prefix(name, "mean_rate"),
+                    format(convertRate(meter.getMeanRate())),
+                    timestamp);
+        }
     }
 
     private void reportHistogram(MetricName name, Histogram histogram, long timestamp) throws IOException {
         final Snapshot snapshot = histogram.getSnapshot();
-        graphite.send(prefix(name, "count"), format(histogram.getCount()), timestamp);
-        graphite.send(prefix(name, "max"), format(snapshot.getMax()), timestamp);
-        graphite.send(prefix(name, "mean"), format(snapshot.getMean()), timestamp);
-        graphite.send(prefix(name, "min"), format(snapshot.getMin()), timestamp);
-        graphite.send(prefix(name, "stddev"), format(snapshot.getStdDev()), timestamp);
-        graphite.send(prefix(name, "p50"), format(snapshot.getMedian()), timestamp);
-        graphite.send(prefix(name, "p75"), format(snapshot.get75thPercentile()), timestamp);
-        graphite.send(prefix(name, "p95"), format(snapshot.get95thPercentile()), timestamp);
-        graphite.send(prefix(name, "p98"), format(snapshot.get98thPercentile()), timestamp);
-        graphite.send(prefix(name, "p99"), format(snapshot.get99thPercentile()), timestamp);
-        graphite.send(prefix(name, "p999"), format(snapshot.get999thPercentile()), timestamp);
+        if (filter.matches(name, histogram, "count")) {
```

Comment:
The issue I have with merging this to v4 is that in v4 all of these lines will have been removed. Instead of each reporter being responsible for selecting which values will be reported for a particular metric, the reporter will either be provided with a collection of measurements or will iterate over the measurements provided by a metric. The effort to update all reports with this logic will be wasted.